### DISCUSSION TITLE
Firmware Settings

### DISCUSSION THREAD

**Discussion by christophepistophe** [winder/Universal-G-Code-Sender#2443] [Help]
Hi. I have just acquired a 3DTek XYZ Carve and am having difficulty getting it going. I have followed the manufacturers instructions and believe I have wired it correctly.
The instructions https://xyzcarve.3dtek.xyz/steps/Step14/ tell me to download UGS Platform, which I have done. I go to 'Setup wizard' and I can connect to my machine via USB. The set up tells me to go to Firmware Settings, update parameter 21 ('Hard Limits Enable') to be 0, Save and Close. I can input 0 as requested but every time I click Save it reverts to 1. When I test the motors in the setup I can get each one to 'Jog' a fraction every time I click. The motors warm up and move so I believe the connection is good but the machine doesn't respond to any further commands until 'Try Homing' . At this stage the Z axis will move but without limitation so it is either driving up on its axis or down into the base board. Either way the only option is to click 'Abort'. The machine does not move on X or Y access.
Any advice would be greatly appreciated please.
Chris

**Comment by breiler** [comment]
> I can input 0 as requested but every time I click Save it reverts to 1. 

This is an old issue with the settings dialog. Type `0` in the edit cell then click somewhere else in the list to exit the cell editor - then click `Save`.

> When I test the motors in the setup I can get each one to 'Jog' a fraction every time I click.

Make sure that the motors are moving in the correct direction in this step. It should do very small step increments:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/d723a084-9d20-430b-928b-d38f8b088f2f)

> At this stage the Z axis will move but without limitation so it is either driving up on its axis or down into the base board.

Invert the homing direction if it is homing in the wrong direction. Make sure that the limit switches is detected in the step prior to homing by triggering them manually with your hand:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/9a35d404-a675-4216-a43d-b90427c3ac71)

**Comment by christophepistophe** [comment]
Hi Joacim

 

Many thanks for your help. I have got past the Firmware settings problem but unfortunately, I am still having problems. I have been through the 3DTek set up and everything seems OK except the limit switches. I bought the machine 2nd hand and the only error I can find is with the limit switches. There are 3 limit switch wires connected to the controller in accordance with the online instructions. There are X and Y switches connected which respond when touched manually although the Z wire in not connected and has been stowed away. I see these as safety cut offs and don’t understand why they prevent me from getting to the homing stage.

 

I have not calibrated the movement yet as this seems like fine tuning when I cant get the motors to function properly.

 

When I get to ‘try homing’ the Z will move in either direction. Having changed field 21 in Firmware Settings this now only moves 10mm or thereabouts. X and Y still do not move. The motors feel too warm to me and I have reset the resistors as per below



 

However, the UGS Screen responds to my commands but the motors, which will jog tiny amounts, will not move.

 

Is there some sort of service I can engage please to help me get going?

 

Many Thanks

 

Chris
